bokehjs/test/test_common/test_canvas.coffee

```diff
@@ -0,0 +1,67 @@
+{expect} = require "chai"
+utils = require "../utils"
+
+canvas = utils.require "common/canvas"
+
+describe "Canvas Model", ->
+
+  beforeEach ->
+    @height = 400
+    @cm = new canvas.Model({'canvas_height': @height, 'canvas_width': 400})
+
+  it "vx_to_sx should return what it is passed", ->
+    x = 12
+    expect(@cm.vx_to_sx(x)).to.eql(x)
+
+  it "vy_to_sy should make y relative to height less 1px", ->
+    view_y = 12
+    expect(@cm.vy_to_sy(view_y)).to.eql(@height - view_y - 1)
+
+  it "v_vx_to_sx should return what it is passed", ->
+    xx = new Float64Array([72.72, 87.87, 224.24, 209.09])
+    expect(@cm.v_vx_to_sx(xx)).to.eql(xx)
+
+  it "v_vy_to_sy should make y relative to canvas height less 1 px", ->
+    view_yy = new Float64Array([10.0, 20.0, 30.0])
+    screen_yy = new Float64Array([@cm.vy_to_sy(view_yy[0]), @cm.vy_to_sy(view_yy[1]), @cm.vy_to_sy(view_yy[2])])
+    expect(@cm.v_vy_to_sy(view_yy)).to.eql(screen_yy)
```

Comment:
Yes I'm just commenting, since I would have expected to see the dilation equally. We could probably investigate it later.